types/float_test.go

```diff
@@ -0,0 +1,86 @@
+package types
+
+import (
+	"database/sql"
+	"github.com/stretchr/testify/require"
+	"testing"
+)
+
+func TestFloat_MarshalJSON(t *testing.T) {
+	subtests := []struct {
+		name   string
+		input  sql.NullFloat64
+		output string
+	}{
+		{"null", sql.NullFloat64{}, `null`},
+		{"invalid", sql.NullFloat64{Float64: 42}, `null`},
+		{"zero", sql.NullFloat64{Float64: 0, Valid: true}, `0`},
+		{"negative", sql.NullFloat64{Float64: -1, Valid: true}, `-1`},
+		{"fraction", sql.NullFloat64{Float64: 0.5, Valid: true}, `0.5`},
+	}
+
+	for _, st := range subtests {
+		t.Run(st.name, func(t *testing.T) {
+			actual, err := Float{st.input}.MarshalJSON()
+
+			require.NoError(t, err)
+			require.Equal(t, st.output, string(actual))
+		})
+	}
+}
+
+func TestFloat_UnmarshalText(t *testing.T) {
+	subtests := []struct {
+		name   string
+		input  string
+		output sql.NullFloat64
+		error  bool
+	}{
+		{"empty", "", sql.NullFloat64{}, true},
+		{"zero", "0", sql.NullFloat64{Float64: 0, Valid: true}, false},
+		{"negative", "-1", sql.NullFloat64{Float64: -1, Valid: true}, false},
+		{"fraction", "0.5", sql.NullFloat64{Float64: 0.5, Valid: true}, false},
+		{"exp", "2e1", sql.NullFloat64{Float64: 20, Valid: true}, false},
```

Comment:
In your other PR testing integers, you had a case with an invalid number, being too big to fit the type. Maybe add something similar here - and in the `TestFloat_UnmarshalJSON` function below - for invalid floats: too big, too much precision, NAN.

Reply:
"too much precision"??

Reply:
A number being more precise as a 64 bit IEEE 754 representation allows. As an inspiration:

```
Python 3.11.9 (main, Apr  2 2024, 08:25:04) [GCC 13.2.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> float("23.12345678901234567890")
23.123456789012344
>>> float(1.0/3.0)
0.3333333333333333
```

Reply:
Sure we wanna cover NaN? 😅 https://www.reddit.com/r/ProgrammerHumor/comments/16bqwkq/only98percentofprogrammerswillunderstand/